src/S3/S3Client.php

```diff
@@ -499,6 +527,66 @@ public static function _apply_use_arn_region($value, array &$args, HandlerList $
         }
     }
 
+    public static function _default_request_checksum_calculation(array $args): string
+    {
+        return ConfigurationResolver::resolve(
+            'request_checksum_calculation',
+            ApplyChecksumMiddleware::DEFAULT_CALCULATION_MODE,
+            'string',
+            $args
+        );
+    }
+
+    public static function _apply_request_checksum_calculation(
+        string $value,
+        array &$args
+    ): void
+    {
+        $value = strtolower($value);
+        if (array_key_exists($value, self::$checksumValidationOptionEnum)) {
+            $args['request_checksum_calculation'] = $value;
+        }
+    }
+
+    public static function _default_response_checksum_validation(array $args): string
+    {
+        return ConfigurationResolver::resolve(
+            'response_checksum_validation',
+            ValidateResponseChecksumResultMutator::DEFAULT_VALIDATION_MODE,
+            'string',
+            $args
+        );
+    }
+
+    public static function _apply_response_checksum_validation(
+        $value,
+        array &$args
+    ): void
+    {
+        $value = strtolower($value);
+        if (array_key_exists($value, self::$checksumValidationOptionEnum)) {
```

Comment:
Same here regarding what happens if the customer provided a value that will not be found in `self::$checksumValidationOptionEnum`?.